### DISCUSSION TITLE
Difficulty running test suites

### DISCUSSION THREAD

**Discussion by JamesPeters86** [pyinstaller/pyinstaller#7464] [Features & Development]
I apologize if this should be in the help section rather than here. It's my first time working on a major project. I'm interested in working on #4482. I decided to run the test suite so I would have something to compare against. The results I got seemed very questionable. I was wondering if there was some setup that I needed to do that's not discussed in the [Dev Guide Quickstart](https://pyinstaller.org/en/stable/development/quickstart.html), [Dev Guide: Running Test Suite](https://pyinstaller.org/en/stable/development/testing.html), or the [tests\README](https://github.com/pyinstaller/pyinstaller/blob/develop/tests/README.md). These issues seemed to mostly reside with the functional tests and they won't run at all if I don't run them alongside the unit tests. What am I doing wrong?

As a side note, \_\_main__.py can't load the PyInstaller module. It works fine if I move it to the toplevel I'm just not sure if that is the "right" thing to do. 

All tests
```
$py.test tests/unit tests/functional -k "not tests/functional/test_libraries.py"
...
== 13 failed, 320 passed, 277 skipped, 68 deselected, 19 xfailed, 28785 warnings, 439 errors in 81.37s (0:01:21) ==
```

Just unit tests 
```
$ py.test tests/unit
...
============= 11 failed, 314 passed, 17 skipped, 11 xfailed, 16101 warnings, 18 errors in 61.34s (0:01:01) ==============
```
Not great, but seemed much more reasonable. 

Just functional tests
```
$py.test tests/functional -k "not tests/functional/test_libraries.py"
ImportError while loading conftest '/home/ubuntu/pyinstaller/tests/functional/conftest.py'.
tests/functional/conftest.py:2: in <module>
    from PyInstaller.utils.conftest import *  # noqa: F401, F403
E   ModuleNotFoundError: No module named 'PyInstaller'
```

**Comment by bwoodsend** [comment]
Urgh, I see that readme hasn't been updated for years. `py.test` is long deprecated. Use `pytest` instead. `ModuleNotFoundError: No module named 'PyInstaller'` likely means that you haven't installed PyInstaller in editable mode (`pip uninstall pyinstaller; pip install -e .`). That should solve your issues with `__main__.py` too.

How far does that get you?

**Reply by JamesPeters86** [reply]
Thanks, that resolved it.
